mypy/typeanal.py

```diff
@@ -232,7 +232,9 @@ def visit_unbound_type(self, t: UnboundType) -> Type:
                     # context. This is slightly problematic as it allows using the type 'Any'
                     # as a base class -- however, this will fail soon at runtime so the problem
                     # is pretty minor.
-                    return AnyType()
+                    any_type = AnyType()
+                    any_type.is_from_silent_import = sym.node.type.is_from_silent_import
```

Comment:
Yes, I was thinking about it. Would probably make it much cleaner. 

What stopped me was that there is already an parameter `implicit` in the constructor from [this commit](https://github.com/python/mypy/commit/6de11e10b3af3ec0b974a7f69664d63f337003ee).
It should be fine if I comment it though.